File: Frontend/src/app/models/Core/birdModel.ts
import { BaseExecIDModel } from "./baseExecIDModel";

export interface BirdModel extends BaseExecIDModel {
    id: string;
    name: string;
    description: string;
    type: string;
    icon: string;
    userID: string;
    birthDate: string | Date;
    createdAt: string | Date;
    updatedAt: string | Date | null;
}

File: API/Plugins/Birds/FeatherTracker.Plugins.Birds/Models/Shared/Birds/BirdModel.cs
using DatabaseSharp.Models;
using FeatherTracker.API.Tools;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace FeatherTracker.Plugins.Birds.Models.Shared.Birds
{
	public class BirdModel : BaseExecIDModel
	{
		[DatabaseSharp(ColumnName = "PK_ID")]
		[JsonPropertyName("id")]
		[Required]
		public Guid ID { get; set; }
		[Required]
		public string Name { get; set; }
		[Required]
		public string Description { get; set; }
		[Required]
		public string Type { get; set; }
		[Required]
		public string Icon { get; set; }
		[DatabaseSharp(ColumnName = "FK_UserID")]
		[Required]
		public Guid UserID { get; set; }

		[DatabaseSharpIgnore(IgnoreAsFill = false)]
		public DateTime CreatedAt { get; set; }
		[DatabaseSharpIgnore(IgnoreAsFill = false)]
		public DateTime? UpdatedAt { get; set; }
	}
}
